### PR TITLE
Add all-in-one Docker Compose and Grafana configuration

### PR DESCRIPTION
**Pull request by window9u**
**What this PR does / why we need it**:

This PR introduces a new `docker-compose-deploy.yml` file that allows running Yorkie, MongoDB, Grafana, and Prometheus together in a single command. While we already have a `docker-compose-full.yml` for running MongoDB, Grafana, and Prometheus, this new file includes the Yorkie server as well. This addition enables easier testing of a complete deployment setup.

Additionally, this PR addresses an issue discovered on Linux (Ubuntu) systems where Prometheus couldn't recognize the `host.docker.internal:8081` and `localhost:8080` targets for Yorkie when using `docker-compose-full.yml` and running the Yorkie server separately. I found that using the Docker bridge IP (typically 172.17.0.1) resolves this issue. A comment explaining this has been added to the `prometheus.yml` file.

However, it's important to note that this solution doesn't work in WSL2. For WSL2 users, it's recommended to use the new `docker-compose-deploy.yml` file instead.

Lastly, I've added Grafana setup instructions to the docker-compose file and included steps for configuring the dashboard in Grafana's README.md.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

1. File naming suggestion:
   - Rename `docker-compose-full.yml` to `docker-compose-observer.yml`
   - Rename `docker-compose-deploy.yml` to `docker-compose-full.yml`

   Although the new file is currently named `docker-compose-deploy.yml`, it doesn't seem entirely suitable for actual deployment purposes. The suggested renaming better reflects the contents and purposes of these files.

2. Grafana dashboard import:

   Initially, I attempted to include dashboard import in the Grafana configuration but found that Grafana doesn't support this feature directly. While it's possible to add JSON files for automatic dashboard setup, I decided against including them in the source code to keep it clean. Instead, I've added instructions for setting up the default dashboard in Grafana's README.md.

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

## Summary by CodeRabbit

- **New Features**
	- Introduced a Docker Compose configuration for deploying Yorkie, MongoDB, Prometheus, and Grafana in a single command.
	- Added comprehensive setup documentation for Grafana, including steps for importing and setting default dashboards.
	- Implemented configuration files for Grafana dashboards and data sources to streamline monitoring and visualization.

- **Documentation**
	- Enhanced README files to provide clearer instructions on deployment and dashboard management.
	- Improved comments in the Prometheus configuration for better user guidance on setting up targets.

### CODE REVIEW

**Comment by krapie** [comment]
@window9u Could you rebase your commit to current upstream? There were some changes in previous commit.

File: build/docker/docker-compose-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        '--mongo-connection-uri',
+        'mongodb://mongo:27017',
+        '--enable-pprof',
+      ]
+    restart: always
+    ports:
+      - '8080:8080'
+      - '8081:8081'
+    depends_on:
+      - mongo
```
Comment (krapie) [inline build/docker/docker-compose-deploy.yml:49]: Add newline below.

FYI: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: build/docker/grafana/dashboard.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+apiVersion: 1
```
Comment (krapie) [inline build/docker/grafana/dashboard.yml:1]: Is this manifest needed even if we need to create/import dashboard ourselves?

File: build/docker/grafana/grafana.ini [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+allow_sign_up = true
+
+[auth.anonymous]
+enabled = false
+
+[dashboards]
+versions_to_keep = 20
+
+[unified_alerting]
+enabled = true
```
Comment (krapie) [inline build/docker/grafana/grafana.ini:21]: > Add newline below.
>
> FYI: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Same here.

**Review by krapie — CHANGES_REQUESTED** [review]
Thank you for your contribution.

1. About docker-compose file

I thought `docker-compose-full.yaml` file was used in our CI process or other places, but it was not. Therefore, as you have suggested, let's integrate into one single file.

2. About folder structure

Since we have created new `grafana` folder, how about integrating `prometheus.yml` file into that folder and rename it to `monitoring` folder?

Also, this approach was used in our devops repository.

3. Grafana dashboard import

If there is no way to do so, I think your instruction is good enough for users to follow.

FYI: In our current chart, we can import Grafana dashboard dynamically with dashboard id.

https://github.com/yorkie-team/yorkie/blob/b77c25af00b23de9e65d61f0a97628471d348d75/build/charts/yorkie-monitoring/values.yaml#L95-L102

File: build/docker/grafana/dashboard.yml (hunk 2 above)
Comment (window9u) [reply build/docker/grafana/dashboard.yml:1]: The dashboard configuration file is not needed if we choose to create or import dashboards manually. Given your question about its significance, I've reconsidered its usefulness. Upon reflection, I agree that the file may not offer substantial advantages. As a result, I've decided to remove it and have committed this change.

**Comment by window9u** [comment]
Thank you for taking the time to review my PR. I appreciate your feedback and have implemented your recommendations!

**Review by krapie — APPROVED** [review]
Thank you for your contribution!

This will definitely allow users to easily provision yorkie server & addons in docker environment 👍🏼

**Pull request merged**